container-deployment/docker-logging-driver.md

```diff
@@ -113,46 +112,109 @@ The application log is stored in the `"log"` field in the record. You can parse
     @type json # apache2, nginx, etc.
   </parse>
 </filter>
+
+<source>
+  @type forward
+  port 24224
+  bind 0.0.0.0
+</source>
+
+<match **>
+  @type stdout
+</match>
 ```
 
-Original Event:
+Then you provide the log message with JSON format:
 
 ```text
-2019-07-22 03:36:39.000000000 +0000 6e8a14315069: {"log":"{\"key\":\"value\"}","container_id":"6e8a1431506936b8568a284f2b0dd4853c250ad85ab7a497f05c4d371f6c3ae6","container_name":"/laughing_beaver","source":"stdout"}
+$ docker run --log-driver=fluentd --log-opt tag=docker ubuntu echo "{\"key\":\"value\"}"
+```
+
+About `--log-opt tag=...`, please refer at [Driver Options](#driver-options) section.
+
+Original Event (without filter plugin):
+
+```text
+2025-02-04 08:20:07.000000000 +0000 docker: {"container_id":"291757f94709abd945ba07c7769c46a5c011f014119c0c56b79e287b00dad4ab","container_name":"/awesome_euler","source":"stdout","log":"{\"key\":\"value\"}"}
 ```
 
 Filtered Event:
 
 ```text
-2019-07-22 03:35:59.395952500 +0000 bac5426337a6: {"container_id":"bac5426337a611fc3b7a0b318c3c45981d2acd80f5c5651088bebb8f1f962583","container_name":"/nostalgic_euler","source":"stdout","log":"{\"key\":\"value\"}","key":"value"}
+2025-02-04 08:20:37.969077831 +0000 docker: {"container_id":"291757f94709abd945ba07c7769c46a5c011f014119c0c56b79e287b00dad4ab","container_name":"/awesome_euler","source":"stdout","log":"{\"key\":\"value\"}","key":"value"}
 ```
 
 ### Additional Step 2: Concatenate Multiple Lines Log Messages
 
 The application log is stored in the `log` field of the record. You can concatenate these logs by using [`fluent-plugin-concat`](https://github.com/fluent-plugins-nursery/fluent-plugin-concat) filter before sending it to the destinations.
 
+At first, you need to create custom docker image due to install the `fluent-plugin-concat` gem in the Fluentd container.
+
```

Comment:
Or directly put filter_concat.rb under plugin directory
(Practically, it requires one file for fluent-plugin-concat)

Reply:
But it is not common way to match all plugins.

Reply:
so add an advice/execuse "one file plugin" use-case as additional note.

Reply:
Added a note
https://github.com/fluent/fluentd-docs-gitbook/pull/563/commits/4fe976578e7e096be08a6e28ab5d4f329323cad0